crates/nu-command/src/platform/is_terminal.rs

```diff
@@ -0,0 +1,89 @@
+use nu_protocol::{
+    ast::Call,
+    engine::{Command, EngineState, Stack},
+    span, Category, Example, PipelineData, ShellError, Signature, Span, Type, Value,
+};
+use std::io::IsTerminal as _;
+
+#[derive(Clone)]
+pub struct IsTerminal;
+
+impl Command for IsTerminal {
+    fn name(&self) -> &str {
+        "is-terminal"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("is-terminal")
+            .input_output_type(Type::Nothing, Type::Bool)
+            .switch("stdin", "Check if stdin is a terminal", Some('i'))
+            .switch("stdout", "Check if stdout is a terminal", Some('o'))
+            .switch("stderr", "Check if stderr is a terminal", Some('e'))
+            .category(Category::Platform)
+    }
+
+    fn usage(&self) -> &str {
+        "Check if stdin, stdout, or stderr is a terminal"
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: r#"Return "terminal attached" if standard input is attached to a terminal, and "no terminal" if not."#,
+            example: r#"if (is-terminal --stdin) { "terminal attached" } else { "no terminal" }"#,
+            result: Some(Value::test_string("terminal attached")),
+        }]
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec![
+            "input", "output", "stdin", "stdout", "stderr", "terminal", "tty",
```

Comment:
```suggestion
            "input", "output", "stdin", "stdout", "stderr", "tty",
```
I don't believe you can have `terminal` as a search_term since the command is named is-terminal